lib/task/import/validate/csvCultureValidator.class.php

```diff
@@ -0,0 +1,146 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * CSV culture column test. Check if present, check values against master list,
+ * and check if piped value.
+ *
+ * @author     Steve Breker <sbreker@artefactual.com>
+ *
+ * @internal
+ * @coversNothing
+ */
+class CsvCultureValidator extends CsvBaseValidator
+{
+    const TITLE = 'Culture Check';
+    protected $cultureColumnPresent;
+    protected $rowsWithBlankCulture = 0;
+    protected $rowsWithPipeFoundInCulture = 0;
+    protected $rowsWithInvalidCulture = 0;
+    protected $invalidCultures = [];
+
+    public function __construct(array $options = null)
+    {
+        $this->setTitle(self::TITLE);
+
+        parent::__construct($options);
+    }
+
+    public function reset()
+    {
+        $this->cultureColumnPresent = null;
+        $this->rowsWithBlankCulture = 0;
+        $this->rowsWithPipeFoundInCulture = 0;
+        $this->rowsWithInvalidCulture = 0;
+        $this->invalidCultures = [];
+
+        parent::reset();
+    }
+
+    public function testRow(array $header, array $row)
+    {
+        parent::testRow($header, $row);
+        $row = $this->combineRow($header, $row);
+
+        // Set if culture column is present.
+        if (!isset($this->cultureColumnPresent)) {
+            $this->cultureColumnPresent = array_key_exists('culture', $row);
+        }
+
+        // If present check contents.
+        if ($this->cultureColumnPresent) {
+            if (empty($row['culture'])) {
+                ++$this->rowsWithBlankCulture;
+            } else {
+                // Check if contains pipe.
+                if (0 < strpos($row['culture'], '|')) {
+                    ++$this->rowsWithPipeFoundInCulture;
+                    $this->testData->addDetail(implode(',', $row));
+
+                    // Keep a list of invalid culture values.
+                    if (!in_array($row['culture'], $this->invalidCultures)) {
+                        $this->invalidCultures[] = $row['culture'];
+                    }
+                }
+                // Validate culture value against AtoM.
+                elseif (!$this->isCultureValid($row['culture'])) {
+                    ++$this->rowsWithInvalidCulture;
+                    $this->testData->addDetail(implode(',', $row));
+
+                    // Keep a list of invalid culture values.
+                    if (!in_array($row['culture'], $this->invalidCultures)) {
+                        $this->invalidCultures[] = $row['culture'];
+                    }
+                }
+            }
+        }
+    }
+
+    public function getTestResult()
+    {
+        if (false == $this->cultureColumnPresent) {
```

Comment:
This can be shortened to `if (!$this->cultureColumnPresent) {`

Reply:
👍